packages/react-scripts/template/README.md

```diff
@@ -850,6 +851,76 @@ In the future we plan to integrate it into Create React App even more closely.
 
 To learn more about Flow, check out [its documentation](https://flow.org/).
 
+## Adding TypeScript
+
+TypeScript is a typed superset of JavaScript that compiles to plain JavaScript.
+
+Recent versions of [TypeScript](https://www.typescriptlang.org/) work with Create React App projects out of the box thanks to Babel 7. Beware that Babel 7 TypeScript does not allow some features of TypeScript such as constant enum and namespaces.
+
+To add TypeScript to a Create React App project, follow these steps:
+
+1. Run `npm install --dev typescript @types/react @types/react-dom @types/jest` (or `yarn add --dev typescript @types/react @types/react-dom @types/jest`).
+2. Rename the `.js` files you want to convert. Use `.tsx` if they use JSX or `.ts` if not (e.g. `git mv src/index.js src/index.tsx`).
+
+3. Create a `tsconfig.json` file at the root directory with the following content:
+
+```json
+{
+  "compilerOptions": {
+    "target": "esnext",
+    "module": "esnext",
+    "moduleResolution": "node",
+    "lib": ["esnext", "dom"],
+    "allowJs": true,
+    "allowSyntheticDefaultImports": true,
+    "emitDecoratorMetadata": false,
+    "esModuleInterop": true,
+    "experimentalDecorators": false,
+    "jsx": "react",
+    "noEmit": true,
+    "skipLibCheck": true
+  }
```

Comment:
I can't really recommend in good faith a config that doesn't also have `"strict": true` 🤔

Reply:
This is missing `isolatedModules: true` which is a requirement for babel's typescript transform. Files that don't compile with `isolatedModules: true` are not guaranteed to compile correctly with babel (and even then there are corner cases like `const enum`s).

This is also missing `baseUrl`.

(actually, I didn't check if CRA lets webpack treat non-relative imports as resolving from the `src` root, but if it does, you need `baseUrl`)

Reply:
I'd also add `noEmitHelpers` and `importHelpers` (requires adding `tslib` as a runtime `dependency`) but it's only needed if tsc's output is used at all.

Reply:
>This is also missing baseUrl.

`baseurl` only works on Babel with `babel-plugin-module-resolver`, doesn't it?

Reply:
That is only if you're using `babel` itself to compile (as in `babel-cli`); as this is going through webpack, webpack is the one responsible for resolution.